newIDE/app/public/External/Utils/pathEditor.js

```diff
@@ -0,0 +1,132 @@
+    cancelButton: 'height:27px;float:right;margin-right:2px;border: 2px solid DeepSkyBlue;border-radius: 1px;margin-top: 5px;background-color:white;',
+    setFolderButton: 'height:27px;float:right;margin-left:2px;margin-right:4px;border: 2px solid DeepSkyBlue;border-radius: 1px;margin-top: 5px;background-color:white;',
+};
+
+let saveFolderLabel,
+    setFolderButton,
+    nameInput,
+    saveButton,
+    cancelButton,
+    fileExistsLabel = null
+
+export let saveOptions = {}
+let projectBasePath;
+export function loadHeader(parentElement, editorContentDocument, saveToGDFunction, cancelChangesFunction, projectPath, initialResourcePath, extension) {
+    projectBasePath = projectPath;
+    if (fs.lstatSync(initialResourcePath).isDirectory()) {
+        initialResourcePath = initialResourcePath + '/NewFile' + extension
+    }
+    saveOptions.folderPath = initialResourcePath.substring(0, initialResourcePath.lastIndexOf('/'));
+    saveOptions.name = initialResourcePath.substring(initialResourcePath.lastIndexOf('/') + 1, initialResourcePath.lastIndexOf('.')); //todo-change with name from path
+    saveOptions.extension = initialResourcePath.substring(initialResourcePath.lastIndexOf('.'), initialResourcePath.length);
+
+    saveFolderLabel = editorContentDocument.createElement('label');
+    saveFolderLabel.addEventListener('click', selectBaseFolderPath);
+    saveFolderLabel.style = headerStyle.saveFolderLabel;
+    saveFolderLabel.textContent = saveOptions.folderPath;
+    parentElement.appendChild(saveFolderLabel);
+
+    nameInput = editorContentDocument.createElement('input');
+    nameInput.oninput = updateBasePath;
+    nameInput.type = 'text';
+    nameInput.style = headerStyle.nameInput;
+    nameInput.value = saveOptions.name;
+    parentElement.appendChild(nameInput);
+
+    fileExistsLabel = editorContentDocument.createElement('label');
+    fileExistsLabel.style = headerStyle.fileExistsLabel;
+    parentElement.appendChild(fileExistsLabel);
+
+    saveButton = editorContentDocument.createElement('button');
+    saveButton.textContent = 'Save';
+    saveButton.style = headerStyle.saveButton;
+    parentElement.appendChild(saveButton);
+    saveButton.addEventListener('click', saveToGDFunction);
+
+    cancelButton = editorContentDocument.createElement('button');
+    cancelButton.textContent = 'Cancel';
+    cancelButton.style = headerStyle.cancelButton;
+    parentElement.appendChild(cancelButton);
+    cancelButton.addEventListener('click', cancelChangesFunction);
+
+    setFolderButton = editorContentDocument.createElement('button');
+    setFolderButton.textContent = 'Set Folder';
+    setFolderButton.style = headerStyle.setFolderButton;
+    parentElement.appendChild(setFolderButton);
+    setFolderButton.addEventListener('click', selectBaseFolderPath);
+
+    updateBasePath()
```

Comment:
As stated earlier, I think this should be put out of the function and called by the same file calling `loadHeader`.